### PR TITLE
fix(jsii): excessive "exclude" in "tsconfig.json"

### PR DESCRIPTION
**Pull request by RomainMuller**
In cases where both `jsii.tsc.outDir` and `jsii.tsc.rootDir` are set, and `rootDir` happens to be a parent directory of `outDir`, the generated `exclude` would match all files from `rootDir`, resulting in an empty input set.

This makes sure to not emit an `exclude` declaration for `outDir` in this case, so that input files do not get excluded.

---

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license].

[Apache 2.0 license]: https://www.apache.org/licenses/LICENSE-2.0

### CODE REVIEW

File: packages/jsii/lib/compiler.ts [hunk 1]
```diff
@@ -289,8 +289,12 @@ export class Compiler implements Emitter {
       ],
       exclude: [
         'node_modules',
-        ...pi.excludeTypescript,
```
Comment (eladb) [inline packages/jsii/lib/compiler.ts:292]: can we write a test for this?

**Review by eladb — COMMENTED** [review]
Ping

**Comment by RomainMuller** [comment]
The PR includes a test by way of jsii-calc-base-of-base... Are you looking for something else?

**Review by eladb — APPROVED** [review]

**Comment by eladb** [comment]
What's up with this PR?

**Pull request merged**